packages/agora-rtc-react/src/hooks/tracks.ts

```diff
@@ -349,3 +355,109 @@ export function useRemoteVideoTracks(
   return tracks;
 }
+
+/**
+ * a hook can create a local video track
+ * unpublish track on unmount.
+ */
+export function useLocalVideoTrack(
+  ready = true,
+  client?: IAgoraRTCClient,
+): ICameraVideoTrack | null {
+  const isConnected = useIsConnected(client);
+  const [track, setTrack] = useState<ICameraVideoTrack | null>(null);
+
+  useAsyncEffect(async () => {
+    if (isConnected && ready && !track) {
+      setTrack(await AgoraRTC.createCameraVideoTrack());
+    }
+    if (!isConnected) {
+      setTrack(null);
+    }
+  }, [isConnected, ready]);
+  return track;
+}
+
+/**
+ * a hook can create a local audio track
+ * close track on unmount.
+ */
+export function useLocalAudioTrack(
+  ready = true,
+  audioTrackConfig: MicrophoneAudioTrackInitConfig = { ANS: true, AEC: true },
+  client?: IAgoraRTCClient,
+): IMicrophoneAudioTrack | null {
+  const isConnected = useIsConnected(client);
+
+  const [track, setTrack] = useState<IMicrophoneAudioTrack | null>(null);
+
+  useAsyncEffect(async () => {
+    if (isConnected && ready && !track) {
+      setTrack(await AgoraRTC.createMicrophoneAudioTrack(audioTrackConfig));
+    }
+    if (!isConnected) {
+      setTrack(null);
+    }
+  }, [isConnected, ready]);
+  return track;
+}
+
+/**
+ * publish tacks when readyToPublish is true
+ * unpublish on unmount.
+ */
+export function usePublish(
+  tracks: (ILocalTrack | null)[],
+  readyToPublish = true,
+  client?: IAgoraRTCClient,
+): void {
+  const resolvedClient = useRTCClient(client);
+  const isConnected = useIsConnected(client);
+  //maintain an internal ref to track published
+  const pubTracks = useRef<(ILocalTrack | null)[]>([]);
+  tracks = tracks.filter(track => track !== null && track !== undefined);
+
+  useAsyncEffect(async () => {
+    if (!resolvedClient || !isConnected || !readyToPublish) {
+      return;
+    }
+    const baseCheck = (_track: ILocalTrack): boolean => {
+      return (
+        // need wait web sdk update
+        // !(resolvedClient["mode"] === "live" && resolvedClient["role"] === "audience")
+        true
+      );
+    };
+    const isPublished = (track: ILocalTrack): boolean => {
+      return pubTracks.current.some(
+        pubTrack => pubTrack && pubTrack.getTrackId() === track.getTrackId(),
+      );
+    };
+    const canPublish = (track: ILocalTrack): boolean => {
+      return baseCheck(track) && track.enabled && readyToPublish && !isPublished(track);
+    };
+    let isSame = true;
+    const newTracks: ILocalTrack[] = [];
+
+    for (let i = 0; i < tracks.length; i++) {
+      const track = tracks[i];
+      if (track) {
+        if (canPublish(track)) {
+          try {
+            await resolvedClient.publish(track);
+          } catch (error) {
+            console.error(error);
+          }
+        }
+        newTracks.push(track);
+        if (isSame) {
+          isSame = i < tracks.length && track.getTrackId() === pubTracks.current[i]?.getTrackId();
```

Comment:
Since `pubTracks` is a ref, there's no need to compute `isSame`.

Reply:
fixed